python/src/trezorlib/cli/firmware.py

```diff
@@ -460,6 +461,36 @@ def upload_firmware_into_device(
         sys.exit(3)
 
 
+def _is_ilu_supported(client: "TrezorClient") -> bool:
+    """Check if the firmware supports interaction-less update."""
+    f = client.features
+    version = (f.major_version, f.minor_version, f.patch_version)
+    return version >= (2, 6, 3)
+
+
+def _is_strict_update(client: "TrezorClient", firmware_data: bytes) -> bool:
+    """Check if the firmware is from the same vendor and the
+    firmware is newer than the currently installed firmware.
+    """
+    try:
+        fw = firmware.parse(firmware_data)
+    except Exception as e:
+        click.echo(e)
+        sys.exit(2)
+
+    if isinstance(fw, firmware.VendorFirmware):
```

Comment:
flip this: if not isinstance-> return false, otherwise continue the checks

Reply:
refactored 0c7a1f94ff272509877c88331208e6a8dfacef34